rtCommon/bidsArchive.py

```diff
@@ -447,6 +475,13 @@ def appendIncremental(self,
             validateAppend: Compares image metadata and NIfTI headers to check
                 that the images being appended are part of the same sequence and
                 don't conflict with each other (default: True).
+            useCache: Put incrementals in a cache before writing to disk. This
+                substantially improves performance when writing a whole run in
+                sequence. However, it can have unexpected results if not used
+                carefully, as querying the archive (e.g., getImages) will not
+                return data in the cache until that data is sent to disk using
+                flushCache. Thus, it is False by default so only intentional use
+                occurs.
```

Comment:
Can getImages() check the cache first before checking disk?

Reply:
Yes -- it was a bit more complex, but I've added in a decorator and a flag that make queries compatible with the cache.